### ISSUE TITLE
Typescript issue on AWS Queue Subscribe Method

### ISSUE THREAD

**Issue by dylancetin** [sst/ion#1057]
Hi,

I have been messing around with sst. It is quite fun. Anyways,

I encountered a type mismatch on aws queue class. I have the following code in my sst config

```ts
    const lambda = new sst.aws.Function("Lambda", {
      handler: "/src/index.handler",
    });

    const queue = new sst.aws.Queue("Queue", {
      fifo: true,
      visibilityTimeout: "10 minutes",
    });

    queue.subscribe(lambda.arn);
```

In my ts server (i have no idea if it is specific to me), i get the following error:
```
     └╴E Argument of type 'Output<string>' is not assignable to parameter of type 'string | FunctionArgs | FunctionArn'.
           Type 'Output<string>' is not assignable to type 'FunctionArn'.
             Type 'Output<string>' is not assignable to type '`arn:${string}`'. typescript (2345) [59, 21]
```

I have for now solved this by doing `queue.subscribe(String(lambda.arn));`

Then i found in [apigatewayv2.ts ](https://github.com/sst/ion/blob/dev/platform/src/components/aws/apigatewayv2.ts#L1020) handler type is defined as "Input<string | FunctionArgs | FunctionArn>"

In [queue.ts](https://github.com/sst/ion/blob/90030e668d353e4a83d71108dbd9f26463fecd06/platform/src/components/aws/queue.ts#L439) subscribe method's subscriber type is defined as    "string | FunctionArgs | FunctionArn". 

So in short i believe it needs to be "Input<string | FunctionArgs | FunctionArn>"

**Comment by dylancetin** [comment]
If it is ok, i would happily open a pr for this

**Comment by jayair** [comment]
Yeah seems like a bug. Would appreciate a PR!

**Comment by dylancetin** [comment]
@fwang i think this will solve my issue can you look into it? 

I have also added Input<> for the similar methods/functions I found in the  other components

**Comment by fwang** [comment]
Thanks @dylancetin, will fix.

Btw just curious, why did u create the function first, and then passing in the arn to `.subscribe()`?

**Comment by dylancetin** [comment]
Not for a particular reason. I was messing around. I tried to make a "multi task" lambda. That would be invokable without sqs if i wanted. Then i gave up and created a new lambda pointing to another handler calling the same function. 

But the tinkering was fun.

**Comment by fwang** [comment]
Fixed in v3.1.44
